Add native fetch client section for remote service consumption (#2519)
cap-js/docs: guides/deploy/to-cf.md

[Learn more about consuming remote services with SAP Cloud SDK.](https://sap.github.io/cloud-sdk/docs/js/overview){.learn-more}

#### Native Fetch Client <Beta /> {#native-fetch}

> As an alternative to SAP Cloud SDK, CAP can use Node.js's built-in `fetch` API for remote service calls. Native fetch is selected per request based on the following rules, in that order:

Reviewer: ```suggestion
As an alternative to SAP Cloud SDK, CAP can use Node.js's built-in `fetch` API for remote service calls. Native fetch is selected per request based on the following rules:
```

> 1. If the destination requires features only available in CloudSDK (e.g., BTP Destination Service resolution or non-basic authentication), CloudSDK is always used.

Reviewer: ```suggestion
1. If the destination requires features only available in SAP Cloud SDK (for example, SAP BTP Destination service resolution or non-basic authentication), the SAP Cloud SDK is always used.
```

Author: Is "SAP BTP Destination service" (service all lowercase) preferred over "SAP BTP Destination Service" (service capitalized)?

The word "service" is part of the name. I would have assumed 'title case' to be used for names. If 'title case' is applied, "service" would have to be capitalized. However, looking at [some of the documentation](https://help.sap.com/docs/connectivity/sap-btp-connectivity-cf/destination-service), within the text bodies, the word "service" is written in lowercase (but not always).

In my opinion title case is easier to read since it allows you to discern that "service" is not part of the text, but part of the name. Which convention are we following?

Reviewer: Within BTP we typically see the "service" as suffix to the product name. Therefore it's only capitalized in titles.

2. If <Config>cds.remote.native_fetch</Config> is explicitly set to `true` or `false`, that setting is used.
3. Otherwise, native fetch is used when `@sap-cloud-sdk/http-client` is not installed.

::: warning Current limitations

> The native fetch client does not yet support named destinations via BTP Destination Service, only [application defined destinations](../services/consuming-services#use-application-defined-destinations). Authentication is limited to `NoAuthentication` and `BasicAuthentication`. CSRF token handling is not yet fully on par with SAP Cloud SDK.

Reviewer: ```suggestion
The native fetch client does not yet support named destinations using the SAP BTP Destination service. Only [application defined destinations](../services/consuming-services#use-application-defined-destinations) are supported. In addition, authentication is limited to `NoAuthentication` and `BasicAuthentication`. CSRF token handling is not yet at parity with SAP Cloud SDK.
```

[Learn more about consuming remote services with SAP Cloud SDK.](https://sap.github.io/cloud-sdk/docs/js/overview){.learn-more}

#### Native Fetch Client <Beta /> {#native-fetch}

> CAP provides a built-in remote client that uses the native Node.js `fetch` API (see limitations below). This means during local development, installing SAP Cloud SDK is not required, though installing and using it remains an option. For production, SAP Cloud SDK is still needed, for example to resolve named destinations via the SAP BTP Destination service.

Reviewer: ```suggestion
CAP provides a built-in remote client that uses the native Node.js `fetch` API (see limitations below). During local development, SAP Cloud SDK is therefore not required, though using it remains an option. For production, SAP Cloud SDK is still needed, for example to resolve named destinations via the SAP BTP Destination service.
```

> The native fetch client is selected per outgoing request based on the following rules:

Reviewer: ```suggestion
CAP selects the native fetch client per outgoing request according to the following rules:
```

> #### Native Fetch Client <Beta /> {#native-fetch}
> 
> CAP provides a built-in remote client that uses the native Node.js `fetch` API (see limitations below). During local development, SAP Cloud SDK is therefore not required, though using it remains an option. For production, SAP Cloud SDK is still needed, for example to resolve named destinations via the SAP BTP Destination service.
> 
> CAP selects the native fetch client per outgoing request according to the following rules:
> 
> 1. If the destination requires features only available in SAP Cloud SDK (for example, SAP BTP Destination service resolution or non-basic authentication), the SAP Cloud SDK is always used.
> 2. If <Config>cds.remote.native_fetch</Config> is explicitly set to `true` or `false`, that setting is used.
> 3. Otherwise, native fetch is used when `@sap-cloud-sdk/http-client` is not installed.
> 
> ::: warning Current limitations
> The native fetch client does not yet support named destinations using the SAP BTP Destination service. Only [application defined destinations](../services/consuming-services#use-application-defined-destinations) are supported. In addition, authentication is limited to `NoAuthentication` and `BasicAuthentication`.
> :::

Reviewer: ```suggestion
#### Native Fetch Client <Beta /> {#native-fetch}

CAP provides a built-in remote client that uses the native Node.js `fetch` API. For limitations, see the warning below. During local development, you don't need SAP Cloud SDK, but you can still use it. For production, you still need SAP Cloud SDK. For example, you use it to resolve named destinations through the SAP BTP Destination service.

CAP selects the native fetch client for each outgoing request according to the following rules:

1. If the destination requires features only available in SAP Cloud SDK (for example, SAP BTP Destination service resolution or non-basic authentication), CAP always uses SAP Cloud SDK.
2. If you explicitly set <Config>cds.remote.native_fetch</Config> to `true` or `false`, CAP uses that setting.
3. Otherwise, CAP uses native fetch when you haven't installed `@sap-cloud-sdk/http-client`.

::: warning Current limitations
The native fetch client does not yet support named destinations using the SAP BTP Destination service. It supports only [application-defined destinations](../services/consuming-services#use-application-defined-destinations). In addition, it limits authentication to `NoAuthentication` and `BasicAuthentication`.
:::
```

> The native fetch client does not yet support named destinations using the SAP BTP Destination service. Only [application defined destinations](../services/consuming-services#use-application-defined-destinations) are supported. In addition, authentication is limited to `NoAuthentication` and `BasicAuthentication`. CSRF token handling is not yet at parity with SAP Cloud SDK.

Author: @sjvans should we remove this?
```suggestion
The native fetch client does not yet support named destinations using the SAP BTP Destination service. Only [application defined destinations](../services/consuming-services#use-application-defined-destinations) are supported. In addition, authentication is limited to `NoAuthentication` and `BasicAuthentication`.
```
